provider/sgx.py

```diff
@@ -0,0 +1,189 @@
+"""
+Module for sgx relevant operations.
+"""
+import logging
+import re
+
+from avocado.utils import process
+from virttest.staging import utils_memory
+from virttest.utils_misc import normalize_data_size
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class SGXError(Exception):
+    """ General SGX error"""
+    pass
+
+
+def _get_epc_size(output):
+    """
+    Get epc size from system message
+
+    :param output: epc section info
+    :return: total epc size
+    """
+    epc_size = 0
+    for line in output.splitlines():
+        tmp_epc = re.search(r'\b0x[0-9a-fA-F]*-0x[0-9a-fA-F]*\b',
+                            str(line))
+        if tmp_epc:
+            epc_size += int(tmp_epc[0].split('-')[1], 16) - int(
+                tmp_epc[0].split("-")[0], 16) + 1
+    return epc_size
+
+
+class SGXTestCapability(object):
+    """
+    Hypervisor sgx test capabilities check.
+    """
+
+    def __init__(self, test, params):
+        """
+
+        :param test: Context of test.
+        :param params: params of running ENV.
+        """
+        self._test = test
+        self._params = params
+
+    def valid_host_sgx_cap(self):
+        """
+        Valid if host sgx enablement
+
+        :return: host epc size/False
+        """
+        try:
+            host_sgx_msg = process.system_output(
+                self._params["host_sgx_check"], shell=True)
```

Comment:
check_cmd = self._params.get('host_sgx_check', 'journalctl|grep -i "sgx: EPC section"'), then we don't have to define host_sgx_check in cfg for other new cases, 

and it's also OK for me to call the command directly here, it's not easy to modify the output(sgx: EPC section) in future I think

But yours is still OK for me

Reply:
Thanks, I agree to remove the host check cmd in cfg, hence totally for a sgx test loop, just need to valid host once.